### PR TITLE
feat: test binding of all services

### PR DESCRIPTION
**Pull request by psturc**
https://issues.jboss.org/browse/AGMS-974
https://issues.jboss.org/browse/AGMS-976
https://issues.jboss.org/browse/AGMS-977

### How to test this
Follow the readme. You can target our OSD cluster

### CODE REVIEW

**Comment by b1zzu** [comment]
:eyes:

File: browser/utils/mdc/bindServiceToApp.js [hunk 1]
```diff
@@ -25,13 +35,37 @@ module.exports = async (appName, serviceName, params) => {
...
+            expect(params).to.include.all.keys('upsAndroidServerKey', 'upsAndroidSenderId')
+            await page.waitForSelector('#root_CLIENT_TYPE')
+            await page.select('#root_CLIENT_TYPE', 'Android')
+            await page.type('#root_platformConfig_googlekey', params.upsAndroidServerKey)
+            await page.type('#root_platformConfig_projectNumber', params.upsAndroidSenderId)
+            break
         default:
-            break;
+            console.error('Unknown service name')
+            process.exit(1)
```
Comment (b1zzu) [inline browser/utils/mdc/bindServiceToApp.js]: It's better to throw an error here
```js
throw new Error('Unknown service name')
```

File: browser/utils/bootstrap.js [hunk 2]
```diff
@@ -21,4 +22,17 @@ before(async () => {
 
 after(() => {
     browser.close()
-})
\ No newline at end of file
+})
+
```
Comment (psturc) [inline browser/utils/bootstrap.js]: Got this from https://github.com/mochajs/mocha/issues/2894#issuecomment-403127932

File: browser/utils/bootstrap.js [hunk 3]
```diff
@@ -21,4 +22,17 @@ before(async () => {
...
+  if (FAILED_TESTS[this.currentTest.file]) {
+    this.skip();
+  }
+});
+
+afterEach(function() {
+  if (this.currentTest.state === "failed") {
+    FAILED_TESTS[this.currentTest.file] = true;
+  }
+});
```
Comment (psturc) [inline browser/utils/bootstrap.js]: Got this from https://github.com/mochajs/mocha/issues/2894#issuecomment-403127932

**Comment by psturc** [comment]
> Just one question: would it be possible to find the MDC_URL without need to pass it manually?
> 
> Otherwise LGTM

I think we would have to use OpenShift/kube client for that, since MDC URL could vary based on what configuration the OpenShift router has. Do you think it's worth implementing?

**Comment by b1zzu** [comment]
> > Just one question: would it be possible to find the MDC_URL without need to pass it manually?
> > Otherwise LGTM
> 
> I think we would have to use OpenShift/kube client for that, since MDC URL could vary based on what configuration the OpenShift router has. Do you think it's worth implementing?

Not now, but if we want to fully automate them we will have to do it, let's open an issue for this, wdyt?

**Comment by psturc** [comment]
> > > Just one question: would it be possible to find the MDC_URL without need to pass it manually?
> > > Otherwise LGTM
> > 
> > 
> > I think we would have to use OpenShift/kube client for that, since MDC URL could vary based on what configuration the OpenShift router has. Do you think it's worth implementing?
> 
> Not now, but if we want to fully automate them we will have to do it, let's open an issue for this, wdyt?

👍  https://issues.jboss.org/browse/AGMS-1111

**Pull request merged**